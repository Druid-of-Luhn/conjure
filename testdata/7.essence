language Essence 2.0

find x,y:int(0..9)

such that
    x = 3*y
