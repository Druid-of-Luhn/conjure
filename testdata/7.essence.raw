Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Letting
                         , "forall"
                         , DeclQuantifier (
                             DeclLambda
                               [ ( "__forall_x"
                                 , TypeBoolean
                                 )
                               , ( "__forall_y"
                                 , TypeBoolean
                                 )
                               ] (
                               GenericNode
                                 And
                                 [ Identifier "__forall_x"
                                 , Identifier "__forall_y"
                                 ] ) ) (
                             DeclLambda
                               [ ( "__forall_x"
                                 , TypeBoolean
                                 )
                               , ( "__forall_y"
                                 , TypeBoolean
                                 )
                               ] (
                               GenericNode
                                 Imply
                                 [ Identifier "__forall_x"
                                 , Identifier "__forall_y"
                                 ] ) ) (
                             ValueBoolean True )
                         )
                       , ( Letting
                         , "exists"
                         , DeclQuantifier (
                             DeclLambda
                               [ ( "__exists_x"
                                 , TypeBoolean
                                 )
                               , ( "__exists_y"
                                 , TypeBoolean
                                 )
                               ] (
                               GenericNode
                                 Or
                                 [ Identifier "__exists_x"
                                 , Identifier "__exists_y"
                                 ] ) ) (
                             DeclLambda
                               [ ( "__exists_x"
                                 , TypeBoolean
                                 )
                               , ( "__exists_y"
                                 , TypeBoolean
                                 )
                               ] (
                               GenericNode
                                 And
                                 [ Identifier "__exists_x"
                                 , Identifier "__exists_y"
                                 ] ) ) (
                             ValueBoolean False )
                         )
                       , ( Letting
                         , "sum"
                         , DeclQuantifier (
                             DeclLambda
                               [ ( "__sum_x"
                                 , TypeInteger
                                 )
                               , ( "__sum_y"
                                 , TypeInteger
                                 )
                               ] (
                               GenericNode
                                 Plus
                                 [ Identifier "__sum_x"
                                 , Identifier "__sum_y"
                                 ] ) ) (
                             DeclLambda
                               [ ( "__sum_x"
                                 , TypeBoolean
                                 )
                               , ( "__sum_y"
                                 , TypeInteger
                                 )
                               ] (
                               GenericNode
                                 Times
                                 [ Identifier "__sum_x"
                                 , Identifier "__sum_y"
                                 ] ) ) (
                             ValueInteger 0 )
                         )
                       , ( Find
                         , "x"
                         , DomainIntegerFromTo (
                             Just ( ValueInteger 0 ) ) (
                             Just ( ValueInteger 9 ) )
                         )
                       , ( Find
                         , "y"
                         , DomainIntegerFromTo (
                             Just ( ValueInteger 0 ) ) (
                             Just ( ValueInteger 9 ) )
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Eq
                      [ Identifier "x"
                      , GenericNode
                          Times
                          [ ValueInteger 3
                          , Identifier "y"
                          ]
                      ]
                  ]
  , metadata = []
  }