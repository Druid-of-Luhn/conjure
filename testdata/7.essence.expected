language Essence 2.0

find x : int(0..9)
find y : int(0..9)

such that
    x = 3 * y
